### PR TITLE
Add `ballerinax/weaviate` connector

### PR DESCRIPTION
**Pull request by nadheesh**
# Description

Address https://github.com/wso2-enterprise/choreo/issues/19562

### CODE REVIEW

File: openapi/weaviate/Ballerina.toml
Comment (AathmanT) [inline openapi/weaviate/Ballerina.toml:3]:
```suggestion
keywords = [ "AI/Weaviate", "Cost/Freemium", "Vector", "Search"]
```

Comment (AathmanT) [reply openapi/weaviate/Ballerina.toml:3]: Shall we use vector databases? and vector search or embedding search

File: openapi/weaviate/Module.md
Comment (AathmanT) [inline openapi/weaviate/Module.md:89]: Shall we add all these codes as a sample so that users can easily use them?
https://github.com/hevayo/ballerina-openai-client/blob/fixes/openai.audio/Module.md?plain=1#L61

File: openapi/weaviate/Ballerina.toml
Comment (AathmanT) [inline openapi/weaviate/Ballerina.toml:8]: Add the icon here also

File: openapi/weaviate/openapi.yaml
Comment (AathmanT) [inline openapi/weaviate/openapi.yaml:1801]: Add new line

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md:11]: Please refer the Package.md template provided in the checklist. https://github.com/ballerina-platform/ballerina-extended-library/discussions/77.

File: openapi/weaviate/Module.md
Comment (praneesha) [inline openapi/weaviate/Module.md:9]:
```suggestion
* Create a Weaviate Cluster using the [Weaviate Cloud Service](https://weaviate.io/developers/weaviate/quickstart/installation#weaviate-cloud-services) or deploy using [Docker/Kubernetes](https://weaviate.io/developers/weaviate/quickstart/installation#running-weaviate-with-docker).
```
Shall we add numbered bullet points here?

File: openapi/weaviate/Module.md
Comment (praneesha) [inline openapi/weaviate/Module.md:37]:
```suggestion
Following is an example of inserting new objects to the Weaviate vector storage as a batch operation.
```

File: openapi/weaviate/Module.md
Comment (praneesha) [inline openapi/weaviate/Module.md:62]:
```suggestion
Once the new records are inserted, you can query the Weaviate vector storage using the [Weaviate GraphQL API](https://weaviate.io/developers/weaviate/api/graphql) similar to the example below.
```

File: openapi/weaviate/Package.md
Comment (praneesha) [inline openapi/weaviate/Package.md:4]:
```suggestion
The `ballerinax/weaviate` package is a [Ballerina](https://ballerina.io/) connector for the Weaviate Vector Search Engine API. The connector provides access to the manipulations of the Weaviate schema, objects, and search vectors based on various criteria. 
```

File: openapi/weaviate/Package.md
Comment (praneesha) [inline openapi/weaviate/Package.md:9]:
```suggestion
| Ballerina language         | 2201.4.1 (Swan Lake Update 4) |
```
Shall we give the link to the Download page here?

File: openapi/weaviate/Package.md
Comment (praneesha) [inline openapi/weaviate/Package.md:10]:
```suggestion
| Weaviate vector eearch API | 1.18.0                        |
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:21]:
```suggestion
# Represents the encoding mechanism details.
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:23]:
```suggestion
    # Defines how multiple values are delimited.
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:25]:
```suggestion
    # Specifies whether arrays and objects should be generated as separate fields.
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:29]:
```suggestion
    # Specifies the custom headers.
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:39]:
```suggestion
# Serialize the record according to the `deepObject` style.
```

File: openapi/weaviate/utils.bal
Comment (praneesha) [inline openapi/weaviate/utils.bal:183]:
```suggestion
# + encodingMap - Details of the serialization mechanism
```

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:184]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:183]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:180]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:167]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:39]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:29]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:25]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:23]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:21]: Since the `utils.bal` file is generated by the tool, let's not change the content.

File: openapi/weaviate/Package.md
Comment (AathmanT) [reply openapi/weaviate/Package.md:11]: This change is suggested by Jo and I think it is better to do that improvement eventhough it's not yet updated in the PR checklist. Anyway we can get other's opinion also.

File: openapi/weaviate/Package.md
Comment (AathmanT) [reply openapi/weaviate/Package.md:9]: Suggested by Jo because Ballerina Central can automatically provide the compatible versions. Even when new releases are available. We can get other's opinion on this too.

File: openapi/weaviate/Ballerina.toml
Comment (oshadajay) [reply openapi/weaviate/Ballerina.toml]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Ballerina.toml
Comment (oshadajay) [reply openapi/weaviate/Ballerina.toml]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/openapi.yaml
Comment (oshadajay) [reply openapi/weaviate/openapi.yaml]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Let's wait until we get the feedback from Ballerina team since I believe there should be consistency among all connectors we have.

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md]: Let's wait until we get the feedback from Ballerina team since I believe there should be consistency among all connectors we have.

File: openapi/weaviate/Package.md
Comment (oshadajay) [reply openapi/weaviate/Package.md:11]: Let's wait until we get the feedback from Ballerina team since I believe there should be consistency among all connectors we have.

File: openapi/weaviate/Module.md
Comment (anupama-pathirage) [inline openapi/weaviate/Module.md:3]:
```suggestion
This is a generated connector for the [Weaviate Vector Search Engine API](https://weaviate.io/developers/weaviate/api) OpenAPI specification. Weaviate is an open-source vector search engine, which allows storing data objects and vector embeddings from the ML models, including the LLMs offered by [OpenAI](https://openai.com/), [Hugging Face](https://huggingface.co/), and [Cohere](https://cohere.ai/). Weaviate provides a powerful GraphQL API for querying the embeddings while looking at the similarity and can scale seamlessly into billions of data objects.
```

File: openapi/weaviate/Module.md
Comment (anupama-pathirage) [inline openapi/weaviate/Module.md:23]:
```suggestion
Create and initialize a `weaviate:Client` with your `Service URL` and the obtained `Authentication key`.
```

File: openapi/weaviate/Module.md
Comment (anupama-pathirage) [inline openapi/weaviate/Module.md:62]:
```suggestion
Once the new records are inserted, you can query the Weaviate vector storage using the [Weaviate GraphQL API](https://weaviate.io/developers/weaviate/api/graphql), similar to the example below.
```

File: openapi/weaviate/Module.md
Comment (anupama-pathirage) [inline openapi/weaviate/Module.md:58]:
```suggestion
    objects: objArr
```

File: openapi/weaviate/Module.md
Comment (anupama-pathirage) [inline openapi/weaviate/Module.md:57]:
```suggestion
weaviate:ObjectsGetResponse[] responseArray = check weaviateClient->/batch/objects.post({
```

File: openapi/weaviate/utils.bal
Comment (nadheesh) [reply openapi/weaviate/utils.bal:27]: @oshadajay we need to report to Nipuna if the generated code/comments have any issues. Those can be bugs

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/27871ccb4e752c477b2ad64c34b99d4c1add24a5.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/27871ccb4e752c477b2ad64c34b99d4c1add24a5.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/27871ccb4e752c477b2ad64c34b99d4c1add24a5.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/27871ccb4e752c477b2ad64c34b99d4c1add24a5.

File: openapi/weaviate/Module.md
Comment (oshadajay) [reply openapi/weaviate/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/27871ccb4e752c477b2ad64c34b99d4c1add24a5.

File: openapi/weaviate/utils.bal
Comment (oshadajay) [reply openapi/weaviate/utils.bal:27]: I have informed them.

**Review by anupama-pathirage — DISMISSED** [review]

File: openapi/weaviate/Package.md
Comment (NipunaRanasinghe) [reply openapi/weaviate/Package.md:11]: Looping in @abeykoon. Could you please share the recommended `module.md`, `pacakge.md` and `README.md` templates for ballerinax packages?

File: openapi/weaviate/utils.bal
Comment (NipunaRanasinghe) [reply openapi/weaviate/utils.bal:27]: @oshadajay @nadheesh created https://github.com/ballerina-platform/openapi-tools/issues/1294 to track the improvement from the tool side. Until we get the fix better not to modify the generated sources manually as @oshadajay suggested

Comment (NipunaRanasinghe) [reply openapi/weaviate/utils.bal:27]: @praneesha FYI

File: openapi/weaviate/Package.md
Comment (NipunaRanasinghe) [reply openapi/weaviate/Package.md]: Looping in @abeykoon. Could you please share the recommended `module.md`, `pacakge.md` and `README.md` templates for ballerinax packages?

File: openapi/weaviate/utils.bal
Comment (nadheesh) [reply openapi/weaviate/utils.bal:27]: ack

File: openapi/weaviate/Module.md
Comment (AathmanT) [reply openapi/weaviate/Module.md]: Yes we can link them

Comment (nadheesh) [reply openapi/weaviate/Module.md]: We will add the links after merging the sample PRs

**Pull request merged**
